Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/CalibrationAPI.cs

```diff
@@ -0,0 +1,389 @@
+
+            Vector3 rodriguesVector = new Vector3(extrinsics[4], extrinsics[5], extrinsics[6]);
+            var angle = Mathf.Rad2Deg * rodriguesVector.magnitude;
+            var axis = rodriguesVector.normalized;
+            Quaternion rightHandedRotation = Quaternion.AngleAxis(angle, axis);
+
+            var rightHandedMatrix = Matrix4x4.TRS(rightHandedPosition, rightHandedRotation, Vector3.one);
+            var leftHandedMatrix = rightHandedMatrix;
+            leftHandedMatrix.m02 *= -1.0f;
+            leftHandedMatrix.m12 *= -1.0f;
+            leftHandedMatrix.m22 *= -1.0f;
+            leftHandedMatrix.m32 *= -1.0f;
+
+            var inverse = leftHandedMatrix.inverse;
+            var inversePosition = inverse.GetColumn(3);
+            var inverseRotation = Quaternion.LookRotation(inverse.GetColumn(2), inverse.GetColumn(1)) * Quaternion.Euler(0, 0, 180);
+
+            var calcExtrinsics = new CalculatedCameraExtrinsics();
+            calcExtrinsics.ViewFromWorld = Matrix4x4.TRS(inversePosition, inverseRotation, Vector3.one);
+            calcExtrinsics.Succeeded = succeeded;
+            return calcExtrinsics;
+        }
+
+        public void PrintLastError()
+        {
+            char[] message = new char[256];
+            if (GetLastErrorMessageNative(message, message.Length))
+            {
+                var output = new string(message);
+                Debug.LogError(output);
+            }
+        }
+
+        public bool ProcessChessboardImage(
+            byte[] image,
+            int imageWidth,
+            int imageHeight,
+            int boardWidth,
+            int boardHeight,
+            byte[] cornersImage = null,
+            byte[] heatmapImage = null,
+            int cornerImageRadias = 3,
+            int heatmapWidth = 12)
+        {
+            cornersImage = cornersImage == null ? new byte[image.Length] : cornersImage;
+            heatmapImage = heatmapImage == null ? new byte[image.Length] : heatmapImage;
+
+            return ProcessChessboardImageNative(
+                image,
+                imageWidth,
+                imageHeight,
+                boardWidth,
+                boardHeight,
+                cornersImage,
+                heatmapImage,
+                cornerImageRadias,
+                heatmapWidth);
+        }
+
+        public List<CalculatedCameraIntrinsics> CalculateChessboardIntrinsics(float chessSquareSize)
+        {
+            int numIntrinsics = 1;
+            int intrinsicsSize = 12;
+            float[] output = new float[numIntrinsics * intrinsicsSize];
+            if (ProcessChessboardIntrinsicsNative(chessSquareSize, output, numIntrinsics))
+            {
+                List<CalculatedCameraIntrinsics> intrinsics = new List<CalculatedCameraIntrinsics>();
+
+                // Note: The undistorted projection matrix is not calculated
+                for (int i = 0; i < numIntrinsics; i++)
+                {
```

Comment:
Nit: It seems a bit odd having a for loop over something of size 1 - if this is because we expect to change this very soon that's fine, but if numIntrisics is only set to 1 to make things more readable (as I assume it is) then probably not doing the for loop would make things clearer about the intent of that variable.

Reply:
Unless I'm missing something here and numIntrinsics is also an outparam for the native API?

---
In reply to: [281373726](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4189#discussion_r281373726) [](ancestors = 281373726)

Reply:
this is old behavior from when i was testing multiple intrinsic calculation processes. For now this should be reduced to one. I likely should remove the num instances for multiple plugin related items.